src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -0,0 +1,121 @@
+            .setStaticReferenceType(settingsVarType)
+            .setMethodName("newBuilder")
+            .build();
+    TypeNode fixedCredentialProvideType =
+        TypeNode.withReference(ConcreteReference.withClazz(FixedCredentialsProvider.class));
+    MethodInvocationExpr credentialArgExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(fixedCredentialProvideType)
+            .setArguments(ValueExpr.withValue(StringObjectValue.withValue("myCredentials")))
+            .setMethodName("create")
+            .build();
+    MethodInvocationExpr credentialsMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(newBuilderMethodExpr)
+            .setArguments(credentialArgExpr)
+            .setMethodName("setCredentialsProvider")
+            .build();
+    MethodInvocationExpr buildMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(credentialsMethodExpr)
+            .setReturnType(settingsVarType)
+            .setMethodName("build")
+            .build();
+    Expr initSettingsVarExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(settingsVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(buildMethodExpr)
+            .build();
+
+    String clientName = JavaStyle.toLowerCamelCase(getClientClassName(service.name()));
+    TypeNode clientType = types.get(getClientClassName(service.name()));
+    VariableExpr clientVarExpr = createVariableExpr(clientName, clientType);
+    MethodInvocationExpr createMethodExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(clientType)
+            .setArguments(settingsVarExpr)
+            .setMethodName("create")
+            .setReturnType(clientType)
+            .build();
+    Expr initClientVarExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(createMethodExpr)
+            .build();
+
+    return writeSampleCode(Arrays.asList(initSettingsVarExpr, initClientVarExpr));
+  }
+
+  // ======================================== Helpers ==========================================//
+
+  private static String getClientClassName(String serviceName) {
+    return String.format(CLASS_NAME_PATTERN, serviceName);
+  }
+
+  private static String getSettingsName(String serviceName) {
+    return String.format(SETTINGS_NAME_PATTERN, serviceName);
+  }
+
+  private static String writeSampleCode(List<Expr> exprs) {
+    List<Statement> statements =
+        exprs.stream().map(e -> ExprStatement.withExpr(e)).collect(Collectors.toList());
+    JavaWriterVisitor visitor = new JavaWriterVisitor();
+    for (Statement statement : statements) {
+      statement.accept(visitor);
+    }
+    return SampleCodeJavaFormatter.format(visitor.write());
+  }
+
+  private static VariableExpr createVariableExpr(String variableName, TypeNode type) {
```

Comment:
What about just using the one-liner instead of this wrapper/helper?

Reply:
This method is used in multiple places in this class. It will be used when compose endpoint sample code.